### ISSUE TITLE
[Feature Suggestion] Replace previous activity with the one just started, if they were both started within N seconds from each other

### ISSUE THREAD

**Issue by motato1** [Medvedev91/timetome#54]
First I would express my gratitude for this incredible app. It took me some time to get the hang of it, but for the last few weeks I've been using it every day and am loving it!

It's amazing to see such a well thought-through and well designed open-source project! 

In my time using timeto.me, I have encountered one issue, though.
I regularly start a wrong activity on accident. While this is not a problem, since changing activities is easy, this leads to plenty of mini-entries in my history which makes reviewing/cleaning it rather cumbersome.

My suggestion is, to have the newer (correct) activity to replace the previous one, if the previous one has not been running for more than X seconds.

E.g.

1. I start activity **A** at **12:30:00**
2. I start activity **B** at **12:30:15**
3. Starting time of activity **B** is set to **12:30:00** and **A**'s record is removed from the history.

I'm open to implement something like this and submit a PR but wanted to discuss this first.

Any comments or suggestions are much appreciated.

**Comment by Medvedev91** [comment]
> It's amazing to see such a well thought-through and well designed open-source project!

Thank you! I'm glad to work on the project that is valuable to the community.

> I regularly start a wrong activity on accident.

This is not the first time users are facing this issue. But different cases require different behaviours.

**Case 1.** Sometimes I accidentally tap twice on tasks in the task list. So replacing just deletes the task, and I can't even remember what it was. It could be an important meeting.

**Case 2.** Sometimes I start a task for a few seconds to mark the task done.

There are other cases where replacing can produce unpredictable behaviour.

Please let me know how do you start tasks on accident? Maybe we can deal with the cause and not the effect.

**Comment by ConEstArthur** [comment]
@Medvedev91 
We can have an option to **ignore time tracked less than xx seconds**. It won't show time tracked less than say 15 seconds in the history.
Which won't change the habit of starting the wrong activity but.. oh well..

But I concur with Case 2 of yours. Sometimes even I would start a task to mark it done.

**Comment by motato1** [comment]
Hi, thank you for your quick reply!

The way I use the app is mainly by using the activities list via the clock button in the bottom left on the main screen. I don't start many activities through tasks.

I have noticed two ways I accidentally start the wrong activities:

1. When I open the activity list to start a timer, I sometimes fat-finger and start one of the pre-set timers (blue numbers) for an activty above or below the one I actually intended to tap.
2. The other way is through the pause feature (the one which starts the "Other" activity). I hit the "||" button to pause my current activity and hit it again to resume. But this doesn't work, as it just starts a new "Other" activity.

I think the second case is just me using the app absent-mindedly. While it's debatable if pause should pause the pause, that's a different issue. I can easily fix it, by paying a bit more attention to how I use the app, until it's become second nature for me.

I am not sure what a *sensible* technical solution for the first one is though.


Also, how is your arm? I hope it's getting better :)

**Comment by ClaymeCall** [comment]
For certain use cases your suggestion could be useful, but for quickly logging stuff I did during the day and placing them at the correct time when I didn't really open the app each time I know i actually prefer the way it works now. 

If it were to be implemented, I think a "task overwrite threshold" setting in seconds would help users adapt the feature to suit their usage. Maybe also a toggle when starting a task,  that enables deletion of the previous one in the instance it was really short.

**Comment by Medvedev91** [comment]
> When I open the activity list to start a timer, I sometimes fat-finger and start one of the pre-set timers (blue numbers) for an activty above or below the one I actually intended to tap.

I think it's a common problem. For me, too. I plan to implement some features to get rid of the need to open the timer sheet. For example, custom buttons on the home screen to start a specified activity with a set timer. I hope this solves the problem.

> The other way is through the pause feature (the one which starts the "Other" activity). I hit the "||" button to pause my current activity and hit it again to resume. But this doesn't work, as it just starts a new "Other" activity.

I just started a beta test for the new pomodoro. It's completely redesigned. I'd like to ask you to test that https://github.com/Medvedev91/timetome/discussions/59

> Also, how is your arm? I hope it's getting better :)

Thank you very much! It gets better every day. 3 weeks left until the cast comes off :)

**Comment by motato1** [comment]
> I think it's a common problem

I've been observing my usage of the app in the recent days and I noticed the problem can probably be solved, or at least alleviated, without any hidden and opinionated mutation of activities.

For me, editing the activity type in the timeline would already be enough I think (see #53). If I could change the currently running activity to a different one, there's no need for auto-merging.

> I just started a beta test for the new pomodoro. It's completely redesigned. I'd like to ask you to test that https://github.com/Medvedev91/timetome/discussions/59

Thank you! I just installed it and will play around and see how the new behavior feels. From first impressions, it is already a big improvement!

> Thank you very much! It gets better every day. 3 weeks left until the cast comes off :)

Glad to hear that :muscle:
